src/wiktextract/extractor/simple/page.py

```diff
@@ -0,0 +1,149 @@
+
+    # from .debug_bypass import debug_bypass
+    # return debug_bypass(wxr, page_title, page_text)
+
+    #####
+
+    # Parse the page, pre-expanding those templates that are likely to
+    # influence parsing; these are problematic templates that generate stuff
+    # that "should" be present for the page to parse better
+    tree = wxr.wtp.parse(
+        page_text,
+        pre_expand=True,
+        additional_expand=ADDITIONAL_EXPAND_TEMPLATES,
+    )
+
+    # print_tree(tree)
+
+    # If this was a normal wiktionary edition, this is where you would split
+    # the page into different language entries; "English", "Gaelic", "Swahili",
+    # etc.
+    # Each section would be a Level 2 node (`NodeKind.LEVEL2`) child of the
+    # page's root node. Wiktionaries do not seem to use LEVEL1 for anything,
+    # although there might be (there definitely is...) an exception somewhere.
+
+    # Some data that is collected is shared among several entries; things like
+    # pronunciation data and etymological data is collected and updated in this
+    # base_data object that can be copied for more specific entries.
+    # Simple English Wiktionary has these sections under Level 3 ("===")
+    # headers, which put them in an awkward hierarchy with the main page and
+    # Part-of-Speech sections, which are higher; they usually appear before
+    # POS sections, but a few pages have a more complex structure. We handle
+    # this, specifically for Simple, by preprocessing the text (above) so that
+    # the page is divided into one or more Level 1 nodes.
+    # If the preprocessor has encountered a structure where you have something
+    # that looks like the start of a section (like "Pronunciation") which comes
+    # after POS data has already started, *and* is followed by a POS section
+    # down the line, that's where we split the page with a Level 1 section.
+    base_data = WordEntry(
+        word=wxr.wtp.title or "ERROR_NO_TITLE",
+        # Simple English wiktionary entries are always for English words
+        lang_code="en",
+        lang="English",
+        pos="ERROR_UNKNOWN_POS",
+    )
+
+    word_datas: list[WordEntry] = []
+
+    flattened_levels = list(tree.find_child_recursively(LEVEL_KIND_FLAGS))
+    headings_of_levels = [
+        clean_node(wxr, None, level.largs[0]).lower()
+        for level in flattened_levels
+    ]
+
+    for i, (heading_title, level) in enumerate(
+        zip(headings_of_levels, flattened_levels)
+    ):
+        # Ignore everything outside of a section with a heading; there shouldn't
+        # be anything there. Previous version of this code looked through that
+        # stuff and would spit out warnings.
+
+        # Collect Etymology (Word parts), Pronunciation data.
+        # Ignore Description for now. XXX maybe implement something if relevant.
+
+        # print(f"=== {heading_title=}")
+
+        if m := POS_STARTS_RE.match(heading_title):
+            pos_data = base_data.copy(deep=True)
```

Comment:
I'd recommend avoid building long regex pattern if possible.

```suggestion
        m = re.search(r"\s*\d+$", heading_title)
        if m is not None:
            heading_title = heading_title[:m.start()]
            pos_num = int(m.group().strip())
        if heading_title in POS_DICTIONARY:
            new_data = process_pos(wxr, base_data.model_copy(deep=True))
            word_datas.append(new_data)
        elif heading_title == "pronunciation":
            process_pron(wxr, level, word_datas[-1] if len(word_datas) > 0 else base_data)
```

New `WordEntry` is only added to `word_datas` if there was a level 2 POS section, and because level 3 can't be the child of next level 2 node, we don't need to check afterward titles.

I think any other cases will all mean the original wikitext are very wrong and we don't need to handle a corrupted layout.

Reply:
This doesn't work because the layout is more complicated.

Reply:
Are there really exists such layout? That will render PronB section under the NounA section. Why would they do that, that doesn't make sense...

Reply:
... I can't find any pages where it happens, except a couple of ones where a section is named wrongly (like in `valiant`, which I've already corrected on Wiktionary). Yet again, I've seen ghosts. :expressionless:

Reply:
By "it happens" I mean that NounC and PronC are that way around, the PronA and PronB stuff is the standard way of doing it on Simple English Wiktionary!

Reply:
The code also works for the non-standard PronC under NounC layout, don't worry too much about rare cases...

I think we'll both agree we can't really expect extract much data from such a small dump file, and its most pages don't have more than two POS sections. I guess simpler and cleaner code as a demonstration and documentation is more important for this edition?

Reply:
The NounC and PronC isn't what happens, only PronA and PronB happens. I've been messing up explaining this.